syntax/crisp_parse.ml

```diff
@@ -97,27 +98,48 @@ let expand_macro_tokens
       | token -> token
   in token_stream_processor token_q wrapper_lexer lexer lexbuf
 
-let print_position outx lexbuf =
-  let pos = lexbuf.lex_curr_p in
-  Printf.fprintf outx "%s:%d:%d" pos.pos_fname
-    pos.pos_lnum (pos.pos_cnum - pos.pos_bol + 1)
+(*FIXME error info should be inside the actual error, not lexbuf*)
+let handle_lex_error ?(silent=false) ex lexbuf =
+  match ex with
+  | Failure _ | Crisp_lexer.Error ->
```

Comment:
Is it possible to differentiate the payload of `Failure`, to avoid coalescing all failures into syntax-related problems?